GUI/QtQuick/TxListModel.cpp

```diff
@@ -319,7 +275,7 @@ void TxListModel::setDetails(const bs::sync::TXWalletDetails& txDet)
    for (int i = 0; i < data_.size(); ++i) {
       const auto& entry = data_.at(i);
       if ((entry.txHash == txDet.txHash) && (txDet.walletIds == entry.walletIds)) {
-         txDetails_[i] = txDet;
+         txDetails_[txId(i)] = txDet;
```

Comment:
txId is not a unique key for transactions, there can be multiple TxDetails for one txId (e.g. when you have 2 wallets and send TX from one of them to another).